Recast/PodcastDetailViewController.swift

```diff
@@ -0,0 +1,165 @@
+        }
+        headerView.titleLabel.text = partialPodcast.collectionName
+        headerView.publisherLabel.text = partialPodcast.artistName
+        headerView.podcastGenres = partialPodcast.genres
+        headerView.filterView.delegate = self
+
+        episodeTableView = UITableView()
+        episodeTableView.backgroundColor = .clear
+        episodeTableView.insetsContentViewsToSafeArea = false
+        episodeTableView.contentInsetAdjustmentBehavior = .never
+        episodeTableView.delegate = self
+        episodeTableView.dataSource = self
+        episodeTableView.rowHeight = UITableView.automaticDimension
+        episodeTableView.tableHeaderView = headerView
+        episodeTableView.register(EpisodeTableViewCell.self, forCellReuseIdentifier: episodeCellReuseIdentifer)
+
+        view.addSubview(backgroundImage)
+        view.addSubview(episodeTableView)
+
+        setUpConstraints()
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        navigationController?.navigationBar.backgroundColor = .clear
+        navigationController?.navigationBar.setBackgroundImage(UIImage(), for: .default)
+        navigationController?.navigationBar.shadowImage = UIImage()
+        navigationController?.navigationBar.isTranslucent = true
+        Podcast.loadFull(from: self.partialPodcast, success: { podcast in
+            self.podcast = podcast
+            self.episodeTableView.reloadData()
+        }) { error in
+            print(error)
+        }
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        navigationController?.navigationBar.backgroundColor = .black
+        navigationController?.navigationBar.isTranslucent = false
+    }
+
+    func setUpConstraints() {
+        // MARK: - Constants
+        let headerViewHeight: CGFloat = 374.5 + UIApplication.shared.statusBarFrame.height
+
+        backgroundImage.snp.makeConstraints { make in
+            make.top.leading.trailing.equalToSuperview()
+            make.height.equalTo(headerViewHeight)
+        }
+
+        headerView.snp.makeConstraints { make in
+            make.top.leading.trailing.equalToSuperview()
+            make.height.equalTo(headerViewHeight)
+            make.width.equalTo(view.safeAreaLayoutGuide)
+            make.bottom.greaterThanOrEqualToSuperview().offset(100)
+        }
+
+        episodeTableView.snp.makeConstraints { make in
+            make.leading.trailing.bottom.equalToSuperview()
+            make.top.equalTo(view.snp.top)
+        }
+
+        episodeTableView.tableHeaderView?.layoutIfNeeded()
+
+        viewDidLayoutSubviews()
+    }
+
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        gradientLayer.frame = backgroundImage.bounds
+    }
+
+    override func didReceiveMemoryWarning() {
+        super.didReceiveMemoryWarning()
+        // Dispose of any resources that can be recreated.
+    }
+
+    func filterEpisodes(filterType: FilterType) {
```

Comment:
```suggestion
    func filterEpisodes(by filterType: FilterType) {
```
to make it more swifty